Documentation/read/query.md

```diff
@@ -0,0 +1,175 @@
+---
+title: Query
+description: About Queries
+keywords: query, ddd, read, queryfor, 
+author: einari, smithmx
+---
+
+# Query
+Looking up collections of data to be presented to users in a view is a common scenario in applications. Dolittle formalizes this through `Query`-types. These types become the contract for your query capabilities. Queries can be optimized for the datasource as you see fit for the feature they're being used in.
+
+## QueryFor
+To implement a query, you basically need to implement the interface `IQueryFor<>`.
+The generic type argument should point to a [read model](read_model.md).
+This is a marker interface and does not require you implement anything specific.
+Instead it is relying on a convention; its looking for a property called `Query`.
```

Comment:
The property must be public, and the return-type must be (at least in the version I'm using) IQueryable<T> where T : IReadModel